### ISSUE TITLE
Can't reproduce the results for GLUE CoLA

### ISSUE THREAD

**Issue by fxmarty** [microsoft/LoRA#35]
My steps:

```
git clone https://github.com/microsoft/LoRA.git
cd LoRA
pip install -e .
cd examples/NLU
pip install -e .
```

Change `export num_gpus=8` to `export num_gpus=1` in `roberta_large_cola.sh`

Then `CUDA_VISIBLE_DEVICES=0 bash roberta_large_cola.sh`

Running on a single A100

Using:
* datasets 2.6.1
* python 3.9.13
* PyTorch 1.13.0+cu117

During training, the `eval_matthews_correlation` is stuck to 0 at all epochs (currently at epoch 7). I actually had the same issue on the current transformers version, and decreasing the learning rate + no warmup helped to regain OKeyish numbers during training, but not as shiny as 0.68.

Do you have an idea of what I could be doing wrong?

**Comment by xijiu9** [comment]
I am facing the similar problem: when I set num_gpu=2 and add gradient_accumulation_steps=4 (which makes the batch size still 32), the average of 5 random seeds on CoLA of roberta-large using LoRA is 67.0. This results are "the result for each run is taken from the best epoch".

**Comment by nbasyl** [comment]
Did anyone know the solution? I am assuming the setting per_device_train_batch_size = 4 on a single GPU is equivalent to total batch size = 4 which is the paper setting, but I am still getting matthews_correlation = 0 during evaluation.

**Comment by MaeChd** [comment]
> Did anyone know the solution? I am assuming the setting per_device_train_batch_size = 4 on a single GPU is equivalent to total batch size = 4 which is the paper setting, but I am still getting matthews_correlation = 0 during evaluation.

Hello, I also encountered the same problem. Did you finally solve it?
